### PR TITLE
build: config_format: Support YAML config format on Windows package

### PR DESCRIPTION
**Pull request by cosmo0920**
Currently, YAML config format is not supported on Windows package.
YAML config format is now actively developed in these days.
We should open up the opportunity for supporting YAML format on Windows.

For backward compatibility, we don't enable this feature when not specifying the directory that contains headers and libraries of libyaml.

----
Enter `[N/A]` in the box, if an item is not applicable to your change.

**Testing**
Before we can approve your change; please submit the following in a comment:
- [ ] Example configuration file for the change
- [ ] Debug log output from testing the change

- [ ] Attached [Valgrind](https://valgrind.org/docs/manual/quick-start.html) output that shows no leaks or memory corruption was found

If this is a change to packaging of containers or native binaries then please confirm it works for all targets.
- [ ] Run [local packaging test](./packaging/local-build-all.sh) showing all targets (including any new ones) build.
- [ ] Set `ok-package-test` label to test for all targets (requires maintainer to do).

**Documentation**

- [ ] Documentation required for this feature

**Backporting**

- [ ] Backport to latest stable release.

----

Fluent Bit is licensed under Apache 2.0, by submitting this pull request I understand that this code will be released under the terms of that license.

### CODE REVIEW

File: .github/workflows/call-build-windows.yaml [hunk 1]
```diff
@@ -127,13 +127,18 @@ jobs:
...
+        shell: cmd
+
       - name: Build Fluent Bit packages
         # If we are using 2.0.* or earlier we need to exclude the ARM64 build as the dependencies fail to compile.
         # Trying to do via an exclude for the job triggers linting errors.
         # This is only supposed to be a workaround for now so can be easily removed later.
         if: ${{ matrix.config.arch != 'amd64_arm64' || needs.call-build-windows-get-meta.outputs.armSupported == 'true' }}
         run: |
-          cmake -G "NMake Makefiles" -DFLB_NIGHTLY_BUILD='${{ inputs.unstable }}' -DOPENSSL_ROOT_DIR='${{ matrix.config.openssl_dir }}' ${{ matrix.config.cmake_additional_opt }} ../
+          cmake -G "NMake Makefiles" -DFLB_NIGHTLY_BUILD='${{ inputs.unstable }}' -DOPENSSL_ROOT_DIR='${{ matrix.config.openssl_dir }}' ${{ matrix.config.cmake_additional_opt }} -DFLB_LIBYAML_DIR=C:\vcpkg\packages\libyaml_${{ matrix.config.vcpkg_triplet }} ../
```
Comment (patrick-stephens) [inline .github/workflows/call-build-windows.yaml:141]: Are these linked statically for the installer?

Comment (cosmo0920) [reply .github/workflows/call-build-windows.yaml:141]: Yes. They are statically linked. No dll dependencies for libyaml are not observed.
fluent-bit_v2 1 3_win64_dev_depds
fluent-bit_v2 1 3_win32_dev_depds

Comment (cosmo0920) [reply .github/workflows/call-build-windows.yaml:141]: For ARM64 Windows on Parallels:
スクリーンショット 2023-05-04 16 04 13

File: cmake/windows-setup.cmake [hunk 2]
```diff
@@ -20,7 +20,9 @@ if(FLB_WINDOWS_DEFAULTS)
   set(FLB_AWS                   Yes)
   set(FLB_HTTP_SERVER           Yes)
   set(FLB_METRICS               Yes)
-  set(FLB_CONFIG_YAML           No)
+  if (NOT FLB_LIBYAML_DIR)
```
Comment (patrick-stephens) [inline cmake/windows-setup.cmake:23]: I'd be tempted to remove it entirely, let's move away from hidden disables and leave the global setting if we can.

**Review by patrick-stephens — APPROVED** [review]

File: cmake/windows-setup.cmake (hunk 2 above)
Comment (cosmo0920) [reply cmake/windows-setup.cmake:23]: Ideally, I'd like to do so. However, AppVeyor build does not handle libyaml on building and libyaml requirements forcibly makes hard to build on Windows for now...

Comment (patrick-stephens) [reply cmake/windows-setup.cmake:23]: Another reason to kill AppVeyor to be honest.
The actual AppVeyor set up is slightly hidden away: https://github.com/fluent/fluent-bit-ci/blob/main/scripts/run-unit-tests.ps1

**Pull request merged**